.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,96 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ develop, master, release/0.8.* ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ develop, master, release/0.8.* ]
```

Comment:
I think we should just use the release prefix and not qualify it as this will need to be maintained when we move to the next minor/major version.

Reply:
Hi Justin, those are good questions. The impetus and goals of this PR were/are:

1. Document what I did when looking into the log4j and other critical security issues I was talking to Shawn about;
2. Look into tools that would look for security issues in our code (as opposed to our dependencies);
3. It seemed easier to show the OpenMRS team how these tools worked via a PR than via a meeting, although that may simply be because I don't like meetings.

As to where this should be targeted, I think the grails 3 branch would be fine. It's already found quite a bit of low-hanging fruit in the grails 1 branch (OBPIH-4215 through OBPIH-4218) so unless we're planning big dependency changes there, it doesn't need to run again there any time soon.

You mentioned "doesn't GitHub already do this for us?" Not being an admin on the OB repo, I don't have a full view of what GitHub does, but I believe these tools do quite a bit more than dependabot.

In terms of cost to port -- They shouldn't be dependent on any grails version, and, I believe, should work just fine on any java or javascript codebase as configured. I think the only time we'd need to edit them would be if/when we move to containers.

Reply:
resolved in c0ab81674